lib/rets/metadata/file_cache.rb

```diff
@@ -0,0 +1,29 @@
+module Rets
+  module Metadata
+
+    # This metadata cache persists the metadata to a file.
+    class FileCache
+
+      def initialize(path)
+        @path = path
+      end
+
+      # Save the metadata.
+      def save(metadata)
+        File.open(@path, "wb") do |file|
+          Marshal.dump(metadata, file)
+        end
+      end
+
+      # Load the metadata.  Return nil if it could not be loaded.
+      def load
+        File.open(@path, "rb") do |file|
+          Marshal.load(file)
+        end
+      rescue IOError, SystemCallError
+        nil
+      end
```

Comment:
:+1: on this gem not being tied to `Marshall` as a serialization method. Caches should use their own serialization logic as they see fit. Whilst this gem does have some (small) existing ties to Marshall, we want to move away from using it in production.

Marshall as a serialization mechanism has many downsides, and few upsides, and by leaving things up to the `Cache` implementation in question, users who want it can have it.